modular_chomp/code/modules/projectiles/ammo_refactor/ammo_override_generator.dm

```diff
@@ -0,0 +1,179 @@
+/*
+
+Leave this unticked by default. This was used to generate ammo_overrides.dm
+
+*/
+
+/obj/item/ammo_magazine/pistol
+/obj/item/ammo_magazine/smg
+/obj/item/ammo_magazine/smg_drum
+/obj/item/ammo_magazine/medium
+/obj/item/ammo_magazine/medium_drum
+/obj/item/ammo_magazine/clip/medium
+/obj/item/ammo_magazine/medium_box
+/obj/item/ammo_magazine/large
+/obj/item/ammo_magazine/large_box
+/obj/item/ammo_magazine/pistol/rubber
+/obj/item/ammo_magazine/pistol/flash
+/obj/item/ammo_magazine/pistol/large
+/obj/item/ammo_magazine/pistol/large/hp
+/obj/item/ammo_magazine/smg/rubber
+/obj/item/ammo_casing/simple/small
+/obj/item/ammo_casing/simple/medium
+/obj/item/ammo_casing/simple/large
+
+/proc/output_all_ammo_types()
+	var/list/types = typesof(/obj/item/ammo_casing)
+	var/listof = list()
+	for(var/type in types)
+		var/obj/item/ammo_casing/casing = new type(null)
+		listof |= casing.caliber
+		qdel(casing)
+	var/output = ""
+	for(var/caliber in listof)
+		output += caliber + "\n"
+	rustg_file_write(output, "data/calibers.txt")
+	types = typesof(/obj/item/weapon/gun/projectile)
+	var/list/magtypes = list()
+	var/list/startmags = list()
+	for(var/type in types)
+		var/obj/item/weapon/gun/projectile/wep = new type(null)
+		magtypes |= wep.allowed_magazines
+		startmags |= wep.magazine_type
+		qdel(wep)
+	output = ""
+	for(var/type in magtypes)
+		output += "[type]\n"
+	output += "\n\n"
+	startmags -= magtypes
+	for(var/type in startmags)
+		output += "[type]\n"
+	output += "\n\n"
+	var/allmags = typesof(/obj/item/ammo_magazine)
+	allmags -= startmags
+	allmags -= magtypes
+	var/secondaryoutput = ""
+	for(var/type in allmags)
+		var/temp = type::parent_type
+		var/tryget = magazine_overrides[type]
+		if(tryget)
+			//output += "[type] = [tryget],\n"
+			continue
+		tryget = magazine_overrides[temp]
+		if(tryget)
+			var/leng = length("[temp]")
+			var/extra = copytext("[type]",leng+1)
+			output += "[type] = [tryget][extra],\n"
+			continue
+
+		secondaryoutput += "[type]\n"
+
+	output += "\n\n"
+	output += secondaryoutput
+	rustg_file_write(output, "data/magazines.txt")
+
+/proc/override_gen()
+	var/list/types = typesof(/obj/item/weapon/gun/projectile)
+	var/list/casing_overrides = list(
+		"small" = /obj/item/ammo_casing/simple/small,
+		"medium" = /obj/item/ammo_casing/simple/medium,
+		"large" = /obj/item/ammo_casing/simple/large,
+	)
+
+	var/list/small_calibers = list(".380", "7.62x25mm", "9x18mm", ".45 LC", "9mm", ".45", "10mm", ".357", ".38", ".44")
+	var/list/medium_calibers = list("5mm caseless", "7.62x54mmR", "7.92x33mm", "9.5x40mm", "7.62x39mm", "5.56x45mm", "7.62mm", "5.45mm", "9x39mm", "mauser", ".48")
+	var/list/large_calibers = list(".338", ".50 BMG", "12.7x108mm", "14.5mm")
+	var/list/override_calibers = small_calibers + medium_calibers + large_calibers
```

Comment:
Okay, in here, we encounter some serious problems. You are more radically changing the game's balance with these three lines.

Previously, .357 and .44, and .45 LC have damage values comparable to medium calibers, and that was on purpose. I noticed .50AE is also missing from here. Pretty sure .50AE is still a thing.

Medium calibers line is just a mess. The difference between many of these calibers was more significant, and lumping them all together is either going to nerf some guns into the ground or elevate guns into being OP depending on how you decided their damage values to get changed. You at least need to have an extra category of ammo here. 5mm and similar calibers were the original medium rifle caliber, with heavy pistol calibers having matching damage values. Whereas 7.62mm stuff (not 7.62x39, that one is definitely medium caliber) was all supposed to be the large rifle caliber. If you're going to make this PR, change this, or else you will muck up the guns a lot more than you intend to.

Large calibers should be elevated to very_heavy_calibers or something to that effect. Like 7.62 is going to put a fat hole in you, but .50 BMG is going to blow a fucking limb off. These calibers are for taking down mechs and shit. Beware however as I think 14.5mm is actually some weird syndicate SMG that is absolutely _not_ the same thing and probably belongs in medium or large calibers and not this category. Otherwise that thing is going to splatter everyone into their atomic components. Or, you accidentally just over-nerfed all the mech killer guns into being absolutely useless.

You need to have a 4th category so some of the medium calibers can be split off and so the true and proper mech-killer calibers can retain their usefulness. All of these are like single shot weapons, or very low capacity weapons that have to be scoped in... except that SMG I mentioned which should definitely not be in there. But maybe that 14.5mm is actually the OG anti-material rifle and I'm just stupid. I genuinely can't remember and I'm in a rush to post this. Just, double-check that before adding it to wherever you put it.